crates/bootstrap/src/lib.rs

```diff
@@ -37,6 +38,169 @@ pub struct ExampleState {
     pub custom: TextureId,
 }
 
+struct App<T: ExampleBody> {
+    yak: Yakui,
+
+    attributes: WindowAttributes,
+    start: Instant,
+
+    state: Option<ExampleState>,
+    window: Option<Window>,
+    app: Option<Graphics>,
+
+    body: T,
+}
+
+impl<T: ExampleBody> ApplicationHandler for App<T> {
+    // This is a common indicator that you can create a window.
+    fn resumed(&mut self, event_loop: &ActiveEventLoop) {
+        let window = event_loop.create_window(self.attributes.clone()).unwrap();
+        window.set_ime_allowed(true);
+
+        let sample_count = get_sample_count();
+
+        let mut app = pollster::block_on(yakui_app::Graphics::new(&window, sample_count));
+
+        // By default, yakui_winit will measure the system's scale factor and pass
+        // it to yakui.
+        //
+        // Sometimes, it might be desirable to scale the UI by a different factor,
+        // like if your game has a "UI scale" option, if you're writing tests, or
+        // you want to ensure your widgets work at a different scale.
+        //
+        // In these examples, setting the YAKUI_FORCE_SCALE environment variable to
+        // a number will override the automatic scaling.
+        if let Some(scale) = get_scale_override() {
+            app.window_mut().set_automatic_scale_factor(false);
+            self.yak.set_scale_factor(scale);
+        }
+
+        // In these examples, set YAKUI_INSET to force the UI to be contained within
+        // a sub-viewport with the given edge inset on all sides.
+        let inset = get_inset_override();
+        if inset.is_some() {
+            app.window_mut().set_automatic_viewport(false);
+        }
+
+        // Preload some textures for the examples to use.
+        let monkey = self
+            .yak
+            .add_texture(load_texture(MONKEY_PNG, TextureFilter::Linear));
+        let monkey_blurred = self
+            .yak
+            .add_texture(load_texture(MONKEY_BLURRED_PNG, TextureFilter::Linear));
+        let brown_inlay = self
+            .yak
+            .add_texture(load_texture(BROWN_INLAY_PNG, TextureFilter::Nearest));
+        let custom = app.renderer.add_texture(
+            custom_texture::generate(&app.device, &app.queue),
+            wgpu::FilterMode::Nearest,
+            wgpu::FilterMode::Nearest,
+            wgpu::FilterMode::Nearest,
+        );
+
+        // Add a custom font for some of the examples.
+        let fonts = self.yak.dom().get_global_or_init(Fonts::default);
+        let font = Font::from_bytes(
+            include_bytes!("../assets/Hack-Regular.ttf").as_slice(),
+            FontSettings::default(),
+        )
+        .unwrap();
+
+        fonts.add(font, Some("monospace"));
+
+        self.state = Some(ExampleState {
+            time: 0.0,
+            monkey,
+            monkey_blurred,
+            brown_inlay,
+            custom,
+        });
```

Comment:
I don't think we need to defer this work to `resumed`, we can do it when we create the yakui state. That also makes it so that `state` doesn't need to be optional.

Reply:
oh right, don't know why we did that, didn't look closely enough probably